src/relay/ir/Relay.g4

```diff
@@ -0,0 +1,140 @@
+grammar Relay;
+
+// Lexing
+// comments
+WS : [ \t\n\r]+ -> skip ;
+LINE_COMMENT : '//' .*? '\n' -> skip ;
+COMMENT : '/*' .*? '*/' -> skip ;
+
+// operators
+MUL: '*' ;
+DIV: '/' ;
+ADD: '+' ;
+SUB: '-' ;
+LT: '<' ;
+GT: '>' ;
+LE: '<=' ;
+GE: '>=' ;
+EQ: '==' ;
+NE: '!=' ;
+
+operatorIdent: CNAME ;
+GLOBAL_IDENT: '@' CNAME ;
+LOCAL_IDENT: '%' CNAME ;
+
+MUT: 'mut' ;
+
+BOOL_LIT
+  : 'true'
+  | 'false'
+  ;
+
+// non-negative floats
+FLOAT
+  : INT '.' INT EXP? // 1.35, 1.35E-9, 0.3, 4.5
+  | INT EXP // 1e10 3e4
+  ;
+
+// non-negative ints
+// INT: '0' | [1-9] DIGIT* ; // no leading zeros
+INT: DIGIT+ ;
+fragment EXP: [eE] [+\-]? INT ; // \- since - means "range" inside [...]
+
+CNAME: ('_'|LETTER) ('_'|LETTER|DIGIT)* ;
+fragment LETTER: [a-zA-Z] ;
+fragment DIGIT: [0-9] ;
+
+// Parsing
+
+// a program is a list of options followed by either an expression or a list of definitions
+// TODO: is this really the desired outline? if there are defns should there be an expression at the end?
+prog: option* (expr | defn+) EOF ;
+
+option: 'set' ident BOOL_LIT ;
```

Comment:
what is option's purpose? Why it is tied to bool literal only?

Reply:
These are intended to be boolean compiler options not none/some options. I think @jroesch can speak more directly to their purpose.

Reply:
In the previous iteration our design goal was to enable the text format to be able to set pragmas/options for the program.